codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieDecoder.java

```diff
@@ -75,7 +75,7 @@ private ServerCookieDecoder(boolean strict) {
         boolean rfc2965Style = false;
         if (header.regionMatches(true, 0, RFC2965_VERSION, 0, RFC2965_VERSION.length())) {
             // RFC 2965 style cookie, move to after version value
-            i = header.indexOf(';') + 1;
+            i = getDelimiterPos(header, 0) + 1;
```

Comment:
> what you mean in your first question?

This code work if `header` starts with the `$Version`:
```java
if (header.regionMatches(true, 0, RFC2965_VERSION, 0, RFC2965_VERSION.length()))
```
not needed to scan `header` from `0`, we can scan immediately from the `"$Version".length()`.

> Second question coment

This comment also is relevant for the current implementation. A `getDelimiterPos` returns `-1` if a `;` is not found, but then it's incremented: `i=-1+1`. Just try execute this:
```java
System.out.println(ServerCookieDecoder.STRICT.decode("$Version=1").size());
System.out.println(ServerCookieDecoder.STRICT.decode("$Version=1;").size());
```
It's looks like a bug,.